docs/admin/mysql-8-upgrade.md

```diff
@@ -0,0 +1,164 @@
+# Migrating to MySQL 8.0
+
+MySQL 5.7 is nearing end of life, so we have moved MISO to MySQL 8.0. To migrate to the latest
+version of MISO, you must switch to MySQL 8.0. Separate instructions are provided below for
+migrating bare metal and Docker Compose setups.
+
+It is recommended to upgrade MISO to the latest 1.x version first using the regular upgrade
+procedure, then use these instructions to upgrade to 2.0.0, then use the regular instructions to
+move to a later 2.x version if available.
+
+## Bare Metal
+
+1. Stop Tomcat to prevent MISO access during maintenance
+1. Make a backup in your usual way, to use for roll-back if necessary
+1. Make a backup of tables only, to use for the migration (views/triggers/functions/procedures will
+  be re-created by Flyway). Replace `lims` with your database name if different
+  
+    ```
+    DATABASE=lims
+
+    mysql -u root -p --skip-column-names -b -e \
+      "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = '${DATABASE}' AND TABLE_TYPE = 'BASE TABLE';" \
+      | xargs mysqldump -u root -p --single-transaction --skip-triggers "${DATABASE}" > misodb_5_7.sql
+    ```
+
+1. remove MySQL 5.7 and install MySQL 8
+   * set the root password. Going forward, root must be used to run Flyway and restore backups
+1. prepare MySQL 8 database. Replace `lims` with your database name if different.
+   * `CREATE DATABASE lims;`
+   * Create any necessary users. e.g.
+
+     ```
+     CREATE USER 'tgaclims'@'localhost' IDENTIFIED BY 'tgaclims';
+     GRANT ALL ON `lims`.* TO 'tgaclims'@'localhost';
+     ```
+
+1. restore tables-only backup to mysql 8.0 using root user
+
+   ```
+   mysql -D ${DATABASE} -uroot -p < misodb_5_7.sql
+   ```
+
+1. Update the database URL in your `ROOT.xml`. Replace `localhost` and `lims` with the correct host
+  and database name. Note the recommended connection options have changed
+
+    ```
+    url="jdbc:mysql://localhost:3306/lims?autoReconnect=true&amp;characterEncoding=UTF-8&amp;allowPublicKeyRetrieval=true&amp;sslMode=DISABLED"
+    ```
+
+1. Follow the usual [update procedure](https://miso-lims.readthedocs.io/projects/docs/en/latest/admin/baremetal-installation-guide/#installing-and-upgrading), but with one additional step:
+  after copying the sqlstore jar, but before running Flyway migrate, run Flyway repair
+
+    ```
+    ./flyway -user=root -password=$MYSQL_ROOT_PASSWORD -url=$MISO_DB_URL -outOfOrder=true -locations=classpath:db/migration,classpath:uk.ac.bbsrc.tgac.miso.db.migration repair -placeholders.filesDir=${MISO_FILES_DIR}
+    ```
+  
+    Note the database URL should be updated (replacing `localhost` and `lims` with the correct host
+    and database name):
+
+    ```
+    jdbc:mysql://localhost:3306/lims?autoReconnect=true&characterEncoding=UTF-8&allowPublicKeyRetrieval=true&sslMode=DISABLED&useSSL=false
+    ```
```

Comment:
I think it' would be valuable to add an explicit instruction here to run the `flyway migrate` step/continue with the migration procedure

Reply:
And might be worth pointing the user to the MySQL 8 timezone docs if they run into "incorrect timezone" issues here

Reply:
Added the instruction. At the step of installing MySQL 8, I added an instruction to  configure the time zone, and noted that you may want to populate the time zone tables